qtconsole/console_widget.py

```diff
@@ -411,6 +414,7 @@ def eventFilter(self, obj, event):
             text widgets.
         """
         etype = event.type()
+        middle_button = QtCore.Qt.MouseButton.MiddleButton if IS_QT6 else QtCore.Qt.MidButton
```

Comment:
What do you think about adding this alias to QtPy instead?

@dalthviz, what do you think? (@dalthviz is the maintainer of QtPy).

Reply:
Checking this with QtPy 2.0.1 seems like the `QtCore.Qt.MidButton` alias for `QtCore.Qt.MouseButton.MiddleButton` is not available with PyQt6 (looking at the code seems like we added the missing mapping for PySide6 but not for PyQt6) however you can use `QtCore.Qt.MiddleButton`  with Qt 5 and 6 (PyQt5/6 and PySide2/6). We probably should add the missing alias for PyQt6 but for the moment changing this line to something like the following should be enought:

```python
        middle_button = QtCore.Qt.MiddleButton
```

Reply:
I think if you use the default channel you will get QtPy 1.11.2 if you use the conda-forge channel you should get the latest version (2.0.1). Could you try to use the conda-forge channel when installing?

Reply:
@dalthviz Thanks for the comments.

I'm a newbie to conda, could you point out what change I would have to make to linux-tests.yml: https://github.com/eyllanesc/qtconsole/blob/ce1373fb83288b8227be4fa1425db5d479727801/.github/workflows/linux-tests.yml to use the conda-forge channel?

Reply:
Glad to help @eyllanesc For the test workflow I think the `setup-miniconda@v2` action has options to set the channels and priority. Something like this in the `with:` section of the step should help with this:

```
channels: conda-forge,defaults
channel-priority: strict
```

Reply:
@dalthviz It did not help many unfortunately, if I use `channel-priority: strict` then the installation process fails: https://github.com/eyllanesc/qtconsole/runs/5483917012?check_suite_focus=true, if I remove it it installs qtpy 1.11.2: https://github.com/eyllanesc/qtconsole/actions/runs/1958675144

Reply:
I think I found the thing here, the channels definition should be done in the `requirements/environment.yml` file itself:

https://github.com/eyllanesc/qtconsole/blob/ce1373fb83288b8227be4fa1425db5d479727801/requirements/environment.yml#L1-L3

So my guess is that you will need to add there the conda-forge channel as first item in the `channels` entry

Reply:
Maybe you could add a constraint in the `environment.yml` for pyqt. Something like `- pyqt >= 5.12`

Reply:
@dalthviz Great, I already corrected other things and now if you install more recent versions of the libraries. Later I will remove redundant code regarding the most recent version of QtPy.